tfhe/src/shortint/server_key/mod.rs

```diff
@@ -57,14 +57,71 @@ impl Display for CheckError {
 
 impl std::error::Error for CheckError {}
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum ShortintBootstrappingKey {
     Classic(FourierLweBootstrapKeyOwned),
-    // TODO: do not serialize ThreadCount, manage that case separately for ser/de and repopulate
-    // on deserialization
     MultiBit(FourierLweMultiBitBootstrapKeyOwned, ThreadCount),
 }
 
+fn zero_thread_count() -> ThreadCount {
+    ThreadCount(0)
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(bound(deserialize = "C: IntoContainerOwned"))]
+enum SerializableShortintBootstrappingKey<C: Container<Element = concrete_fft::c64>> {
+    Classic(FourierLweBootstrapKey<C>),
+    MultiBit(
+        FourierLweMultiBitBootstrapKey<C>,
+        #[serde(skip_serializing, default = "zero_thread_count")] ThreadCount,
+    ),
+}
+
```

Comment:
It doesn't seem to me as though `SerializableShortintBootstrappingKey::MultiBit` actually need the `ThreadCount` member, or am I missing something ?
Removing it could clear up this enum definition

Reply:
fixed ! much better indeed, not sure what I was thinking